tap_dashboard_reports/period.py

```diff
@@ -78,3 +101,8 @@ def last_day_of_month(dt):
 
 def first_day_of_month(dt):
     return dt.replace(day=1)
+
+
+def previous_day_of_week(dt):
```

Comment:
If `dt` is Monday the result of this function will be the next Monday, but should be Sunday, the last day of the week